Scene/scene.js

```diff
@@ -56,22 +56,34 @@ class Scene {
      * @returns {number} identifier for model in the scene
      */
     async AddToScene(source, start, end, cadence, scale, layer_order) {
-        let sun = await ModelFactory.CreateSolarModel(source, start, end, cadence, scale);
-        let model = await sun.GetModel();
-        this._scene.AddModel(model);
-
-        let id = this._count++;
-        this._models[id] = {
-            model: sun,
-            order: layer_order,
-        };
-        sun.SetTime(this._current_time);
-        if (this._count == 1) {
-            this._scene.MoveCamera(sun.GetObserverPosition());
-            this._scene.PointCamera(await sun.GetPosition());
+        try {
+            // Start the loading animation
+            Loader.start();
+            let sun = await ModelFactory.CreateSolarModel(source, start, end, cadence, scale);
+            let model = await sun.GetModel();
+            this._scene.AddModel(model);
+    
+            let id = this._count++;
+            this._models[id] = {
+                model: sun,
+                order: layer_order,
+            };
+            sun.SetTime(this._current_time);
+            if (this._count == 1) {
+                this._scene.MoveCamera(sun.GetObserverPosition());
+                this._scene.PointCamera(await sun.GetPosition());
+            }
+            this._SortLayers();
+            // End the loading animation
+            Loader.stop();
+            return id;
+        } catch (e) {
+            Loader.stop();
+            console.error(e);
+            // TODO: Use a nicer error method than alert
+            alert("Couldn't add source to the scene!");
```

Comment:
If we just return here, then the catch block won't fire in `UI/source_controls.js` and it will continue processing with `undefined` as the return value

```suggestion
            throw e;
```

We can use the catch block to stop the loader, and then re-throw the error so the handling in `source_control.js` will execute.